cocotb/regression.py

```diff
@@ -396,7 +391,17 @@ def _log_sim_summary(self):
         summary += "*************************************************************************************\n"
 
         self.log.info(summary)
-
+    
+    def _safe_divide(self, sim_time_ns, real_time):
+        try:
+            ratio_time  = sim_time_ns / real_time
+        except ZeroDivisionError:
+            if round(sim_time_ns, 2) == 0:
+                ratio_time = float('nan')
+            else:
+                ratio_time = float('inf')
+        return ratio_time
```

Comment:
Perhaps better as
```python
@staticmethod
def _time_ratio(sim_time_ns, real_time):
    try:
        return sim_time_ns / real_time
    except ZeroDivisionError:
        if round(sim_time_ns, 2) == 0:
            return float('nan')
        else:
            return  float('inf')
```

Reply:
You are completely right. Thank you @eric-wieser !